### PR TITLE
parquet: fix ValidateDatum to handle BIT(0) arrays

### LINKED ISSUE #137757: parquet: decoder should handle empty bit(0) arrays
sql/importer.TestRandomParquetExports [failed](https://tanzanite.cluster.engflow.com/invocations/default/1e04a3aa-13a4-4cca-b923-2eeaffc4fadd?testReportRun=2&testReportShard=1&testReportAttempt=1#targets-Ly9wa2cvc3FsL2ltcG9ydGVyOmltcG9ydGVyX3Rlc3Q=) on release-24.3 @ [112484d48f818e6e8585fa944a150d06e5acbbfb](https://github.com/cockroachdb/cockroach/commits/112484d48f818e6e8585fa944a150d06e5acbbfb):

```
=== RUN   TestRandomParquetExports
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestRandomParquetExports1839873803
    test_log_scope.go:76: use -show-logs to present logs inline
    test_server_shim.go:152: automatically injected an external process virtual cluster under test; see comment at top of test_server_shim.go for details.
    exportparquet_test.go:252: exporting as parquet from random table with schema: 
         [[table3 CREATE TABLE public.table3 (
        	col3_0 INT8 NOT NULL,
        	col3_1 INT8 NOT NULL,
        	col3_2 BIT(0)[] NOT NULL,
        	col3_3 BPCHAR NOT NULL,
        	col3_4 DECIMAL NOT NULL,
        	col3_6 DECIMAL NOT NULL,
        	col3_7 INT8 NOT NULL AS (abs(col3_1)) VIRTUAL,
        	CONSTRAINT table3_pkey PRIMARY KEY (col3_6 DESC, col3_2 DESC, col3_3 ASC),
        	FAMILY fam_0_col3_0_col3_1 (col3_0, col3_1),
        	FAMILY fam_1_col3_2 (col3_2),
        	FAMILY fam_2_col3_4 (col3_4),
        	FAMILY fam_4_col3_6_col3_3 (col3_6, col3_3)
        )]]
    exportparquet_test.go:168: 
        	Error Trace:	pkg/sql/importer/exportparquet_test.go:168
        	            				pkg/sql/importer/exportparquet_test.go:134
        	            				pkg/sql/importer/exportparquet_test.go:121
        	            				pkg/sql/importer/exportparquet_test.go:262
        	Error:      	Not equal: 
        	            	expected: &tree.DBitArray{BitArray:bitarray.BitArray{words:[]uint64{}, lastBitsUsed:0x0}}
        	            	actual  : &tree.DBitArray{BitArray:bitarray.BitArray{words:[]uint64(nil), lastBitsUsed:0x0}}
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -2,4 +2,3 @@
        	            	  BitArray: (bitarray.BitArray) {
        	            	-  words: ([]uint64) {
        	            	-  },
        	            	+  words: ([]uint64) <nil>,
        	            	   lastBitsUsed: (uint8) 0
        	Test:       	TestRandomParquetExports
    panic.go:626: -- test log scope end --
test logs left over in: outputs.zip/logTestRandomParquetExports1839873803
--- FAIL: TestRandomParquetExports (15.81s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>deadlock=true</code>
 - <code>run=2</code>
 - <code>shard=1</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #137673 sql/importer: TestRandomParquetExports failed [C-test-failure O-robot T-sql-queries branch-release-24.2 release-blocker]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRandomParquetExports.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

Jira issue: CRDB-45732

### PR DESCRIPTION
**Pull request by rharding6373**
The parquet test utility's ValidateDatum used deep equality to compare DBitArray values. For BIT(0) columns, the SQL layer produces a BitArray with a non-nil empty words slice, but the parquet decoder's bitarray.Parse("") returns a nil words slice. These are semantically equivalent but fail deep equality. Compare using string representation instead, matching the pattern used for other types like JSON and Date.

Also remove the workaround in TestRandomParquetExports that skipped tables with BIT(0) columns, and add a dedicated test case for empty bit array round-tripping.

Fixes #137757

### CODE REVIEW

File: pkg/util/parquet/writer_test.go [hunk 1]
```diff
@@ -324,6 +324,18 @@ func TestBasicDatums(t *testing.T) {
 				}, nil
 			},
 		},
+		{
+			name: "bitarray_empty",
```
Comment (aerfrei) [inline pkg/util/parquet/writer_test.go:334]: This test as written doesn't actually catch the regression. It seems that bitarray.MakeZeroBitArray(0) actually writes a nil array to parquet so isn't the case that was at issue. When pressed claude suggested:
```
		{
			name: "bitarray_empty",
			sch: &colSchema{
				columnTypes: []*types.T{types.MakeBit(0), types.MakeBit(0)},
				columnNames: []string{"a", "b"},
			},
			datums: func() ([][]tree.Datum, error) {
				// Create BIT(0) with non-nil empty slice to reproduce the bug.
				// SQL creates BitArray{words: []uint64{}, lastBitsUsed: 0} (non-nil empty)
				// but parquet decoder's bitarray.Parse("") returns nil words.
				ba, err := bitarray.FromEncodingParts([]uint64{}, 0)
				if err != nil {
					return nil, err
				}
				return [][]tree.Datum{
					{&tree.DBitArray{BitArray: ba}, tree.DNull},
				}, nil
			},
		},
```
Which works.

File: pkg/util/parquet/testutils.go [hunk 2]
```diff
@@ -419,6 +419,12 @@ func ValidateDatum(t *testing.T, expected tree.Datum, actual tree.Datum) {
 		require.Equal(t, expected.(*tree.DEnum).LogicalRep, actual.(*tree.DEnum).LogicalRep)
 	case types.CollatedStringFamily:
 		require.Equal(t, expected.(*tree.DCollatedString).Contents, actual.(*tree.DCollatedString).Contents)
+	case types.BitFamily:
+		// Compare using string representation to avoid nil vs empty slice
+		// mismatch. BIT(0) produces a BitArray with an empty words slice from
+		// SQL, but the parquet decoder's bitarray.Parse("") returns a nil words
+		// slice. Both are semantically equivalent.
+		require.Equal(t, expected.String(), actual.String())
```
Comment (aerfrei) [inline pkg/util/parquet/testutils.go:427]: I was curious if this would result in this check being looser but seems like these strings are equivalent to the ones in the underlying parquet representation. So, this should be full equivalence as far as we're concerned. This is a fine change.

**Review by aerfrei — CHANGES_REQUESTED** [review]

File: pkg/util/parquet/writer_test.go (hunk 1 above)
Comment (rharding6373) [reply pkg/util/parquet/writer_test.go:334]: Nice catch! Sorry for the AI slop. It's fixed now.

File: pkg/util/parquet/testutils.go (hunk 2 above)
Comment (rharding6373) [reply pkg/util/parquet/testutils.go:427]: Thanks for checking this.

**Comment by rharding6373** [comment]
Thank you for the review! Apologies for the AI slop, I'll try to be more careful letting bugs like that through in the future.

**Review by aerfrei — APPROVED** [review]
